### PR TITLE
Add initial bazel build rules

### PR DESCRIPTION
**Pull request by DanG100**
These rules are working, will add to CI in future PR

### CODE REVIEW

**Review by wenovus — COMMENTED** [review]
How does this affect repositories that import lemming?

File: proto/dataplane/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    deps = ["//proto/forwarding"],
+)
+
+go_library(
+    name = "dataplane",
+    srcs = ["generate.go"],
+    embed = [":dataplane_go_proto"],
+    importpath = "github.com/openconfig/lemming/proto/dataplane",
+    visibility = ["//visibility:public"],
+)
```
Comment (wenovus) [inline proto/dataplane/BUILD.bazel:26]: This looks like a duplicate of `dataplane_go_proto`?

File: patches/BUILD.bazel
[the remark this answers is not in the document]
Comment (DanG100) [reply patches/BUILD.bazel:1]: That patches (diffs) applied to repos (gnsi, gnoi, etc) to fix some of the build rules in those repos. I intend to upstream those change later.

File: proto/dataplane/BUILD.bazel (hunk 1 above)
Comment (DanG100) [reply proto/dataplane/BUILD.bazel:26]: they are actually subtly different, these rules are generated automatically.

File: .bazelrc [hunk 2]
```diff
@@ -0,0 +1 @@
+build --cxxopt=-std=c++14 --host_cxxopt=-std=c++14
```
[the remark this answers is not in the document]
Comment (DanG100) [reply .bazelrc]: changed to 17

**Comment by DanG100** [comment]
> How does this affect repositories that import lemming?

We will make sure the lemming is usable with standard go tooling. I'll include some instructions in a follow up PR. But it's pretty simple, just need to keep go.mod and generate pb.go files up to date.

**Review by wenovus — APPROVED** [review]

**Pull request merged**